build_runner/lib/src/generate/build_impl.dart

```diff
@@ -122,6 +122,8 @@ class BuildImpl {
   final bool _verbose;
   final BuildEnvironment _environment;
 
+  BuildPerformanceTracker _performanceTracker;
```

Comment:
non-final fields in this class make me a little nervous... I'm not sure I have a better suggestion though and at least its private 🤷‍♂️

Reply:
Ya, it isn't ideal. We actually have a lot of these that have to get cleaned up after each build (_lazyPhases, _resourceManager).

The real solution is probably to create another class that handles running a single build, but we can do that separately.